mkdocs/utils/__init__.py

```diff
@@ -431,7 +435,10 @@ def nest_paths(paths):
 
         branch = nested
         for part in parts:
-            part = filename_to_title(part)
+            dir = False
+            if os.path.basename(part) != path:
+                dir = True
```

Comment:
Why not change the line above from `!=` to `==` and then change this line to:

```python
part = os.path.splitext(part)[0]
```

Then you can remove the call to `os.path.splitext` in `filename_to_title` and the new `dir` parameter becomes unnecessary.